pkg/bgpv1/cell.go

```diff
@@ -56,3 +58,24 @@ func newBGPPeeringPolicyResource(lc hive.Lifecycle, c client.Clientset, dc *opti
 			c.CiliumV2alpha1().CiliumBGPPeeringPolicies(),
 		), resource.WithMetric("CiliumBGPPeeringPolicy"))
 }
+
+func newSecretResource(lc hive.Lifecycle, c client.Clientset, dc *option.DaemonConfig) resource.Resource[*slim_core_v1.Secret] {
```

Comment:
@dylandreimerink I need your insight here. Should we move this Secret resource to the shared resource types?